src/smtp_server.c

```diff
@@ -123,59 +139,82 @@ int smtp_server_response(bfd_t *f, int code, const char *message)
 int smtp_server_process(struct smtp_server_context *ctx, const char *cmd, const char *arg, bfd_t *stream)
 {
 	int schs, continue_session = 1;
+	int hdlr_idx;
+	int code;
+
 	struct smtp_cmd_hdlr_list *hlink;
+	struct smtp_cmd_hdlr *cmd_hdlr;
+	char *message;
 
 	do {
-		/* Save ctx->node to local var, since it can be changed by a
-		 * command handler *while* we are walking the handler list */
-		struct smtp_cmd_tree *node = ctx->node;
-
-		ctx->code = 0;
-		ctx->message = ctx->prev_message = NULL;
+		code = 0;
+		message = NULL;
+	    hdlr_idx = smtp_get_hdlr_idx(cmd);
```

Comment:
Pay attention to indentation. Configure your editor to always use tab characters.